### ISSUE TITLE
`patch_content` breaks Markdown structure when appending content to headings

### ISSUE THREAD

**Issue by 7shi** [MarkusPfundstein/mcp-obsidian#5]
When using `patch_content` to append content to a heading section, it doesn't add necessary line breaks, which breaks the Markdown structure.

Example:

When applying:
```javascript
{
  operation: "append",
  target_type: "heading",
  target: "test1",
  content: "baz"
}
```

to:
```markdown
# test1
foo
# test2
bar
```

Current output:
```markdown
# test1
foo
baz# test2
bar
```

Expected output:
```markdown
# test1
foo
baz
# test2
bar
```

The function should automatically handle line breaks to preserve Markdown formatting.

**Comment by MarkusPfundstein** [comment]
We need to check what the API expects here and what is the behaviour of the API.

**Comment by 7shi** [comment]
Yes, I agree. That's why I avoided creating a PR independently and raised this as an issue first.
